### DISCUSSION TITLE
Build on Apple M2 Max (Silicon chip)

### DISCUSSION THREAD

**Discussion by mourad-ghafiri** [rabbitmq/rabbitmq-objc-client#239] [General]
### Describe the bug

Cartfile:

github "rabbitmq/rabbitmq-objc-client" "v0.12.0"

carthage bootstrap

*** Checking out CocoaAsyncSocket at "7.6.5"
*** Checking out JKVValue at "v1.3.3"
*** Checking out rabbitmq-objc-client at "v0.12.0"
*** xcodebuild output can be found in /var/folders/yl/f216700511jfcjdmvnx25sn40000gn/T/carthage-xcodebuild.WHLMCN.log
*** Building scheme "iOS Framework" in CocoaAsyncSocket.xcodeproj
A shell task (/usr/bin/xcrun lipo -create /Users/workspace/Library/Caches/org.carthage.CarthageKit/DerivedData/16.0_16A242d/CocoaAsyncSocket/7.6.5/Build/Intermediates.noindex/ArchiveIntermediates/iOS\ Framework/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/CocoaAsyncSocket.framework/CocoaAsyncSocket /Users/workspace/Library/Caches/org.carthage.CarthageKit/DerivedData/16.0_16A242d/CocoaAsyncSocket/7.6.5/Build/Products/Release-iphonesimulator/CocoaAsyncSocket.framework/CocoaAsyncSocket -output /Users/workspace/Desktop/dependencies/Carthage/Build/iOS/CocoaAsyncSocket.framework/CocoaAsyncSocket) failed with exit code 1:
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: /Users/workspace/Library/Caches/org.carthage.CarthageKit/DerivedData/16.0_16A242d/CocoaAsyncSocket/7.6.5/Build/Intermediates.noindex/ArchiveIntermediates/iOS Framework/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/CocoaAsyncSocket.framework/CocoaAsyncSocket and /Users/workspace/Library/Caches/org.carthage.CarthageKit/DerivedData/16.0_16A242d/CocoaAsyncSocket/7.6.5/Build/Products/Release-iphonesimulator/CocoaAsyncSocket.framework/CocoaAsyncSocket have the same architectures (arm64) and can't be in the same fat output file

Building universal frameworks with common architectures is not possible. The device and simulator slices for "CocoaAsyncSocket" both build for: arm64
Rebuild with --use-xcframeworks to create an xcframework bundle instead.

### Reproduction steps

1. create Cartfile with conent : github "rabbitmq/rabbitmq-objc-client" "v0.12.0"
2. Run carthage bootstrap

### Expected behavior

build the lib correctly to be used in a swift project.

**Comment by michaelklishin** [comment]
@mourad-ghafiri you are welcome to investigate what's going on. This project predates Apple Silicon by many years. https://github.com/rabbitmq/rabbitmq-objc-client/issues/231 might be relevant.
